frontend/src/features/books/BookDetails.jsx

```diff
@@ -30,13 +25,33 @@ function BookDetails() {
     fetchCurrentBook()
   }, [id])
 
-  if (!title) return <h3>Loading...</h3>
+  const deleteBook = async () => {
+    try {
+      const response = await fetch(`${API_URL}/${id}`, {
+        method: "DELETE"
+      })
+      if (response.ok) {
+        navigate('/')
+      } else {
+        const errorText = await response.text()
+        throw new Error(`Error ${response.status}: ${errorText}`)
+      }
+    } catch (e) {
+      console.log("An error occurred", e)
+    }
+  }
+
+  if (!book) return <h3>Loading...</h3>
```

Comment:
@nshelokar-vf, why we are showing loading when a book is not present?

Reply:
Sir I thought It would give better user experience. But if it is wrong I will remove it.